scripts/index-genenetwork

```diff
@@ -208,35 +208,83 @@ def locked_xapian_writable_database(path: pathlib.Path) -> xapian.WritableDataba
         db.close()
 
 
-def build_rdf_cache(sparql_uri: str, query: str, remove_common_words: bool = False):
-    cache = {}
+def build_monadic_dict(dic: dict) -> MonadicDict:
+    """Recursively build a monadicdict, only handles list and dicts"""
+
+    def handle_list(original_list: list) -> List[Any]:
+        new_list = []
+        for item in original_list:
+            new_item = item
+            if isinstance(item, list):
+                new_item = handle_list(item)
+            elif isinstance(item, dict):
+                new_item = build_monadic_dict(item)
+            new_list.append(Just(new_item))
+        return new_list
+
+    res = MonadicDict()
+    for key, value in dic.items():
+        if isinstance(value, dict):
+            res[key] = Just(build_monadic_dict(value))
+        elif isinstance(value, list):
+            res[key] = Just(handle_list(value))
+        else:
+            res[key] = Just(value)
+    return res
+
+
+def get_count_and_cache_from_rdf_bindings(
+    bindings: list,
+) -> Tuple[Counter, MonadicDict]:
+    cache = MonadicDict()
+    count: Counter[str] = Counter()
+    words_regex = re.compile(r"\w+")
+    for entry in bindings:
```

Comment:
@jnduli how you build the monadic dict IMHO is a tad bit too complicated for me to follow.  I would suggest that from the returned bindings, build a dict first, then do the thing with the counter, and finally return the cache as a monadic dict as: `MonadicDict(my_foo_dict)`